### PR TITLE
[SPARK-15827][BUILD] Publish Spark's forked sbt-pom-reader to Maven Central

### PR DESCRIPTION
**Pull request by JoshRosen**
Spark's SBT build currently uses a fork of the sbt-pom-reader plugin but depends on that fork via a SBT subproject which is cloned from https://github.com/scrapcodes/sbt-pom-reader/tree/ignore_artifact_id. This unnecessarily slows down the initial build on fresh machines and is also risky because it risks a build breakage in case that GitHub repository ever changes or is deleted.

In order to address these issues, I have published a pre-built binary of our forked sbt-pom-reader plugin to Maven Central under the `org.spark-project` namespace and have updated Spark's build to use that artifact. This published artifact was built from https://github.com/JoshRosen/sbt-pom-reader/tree/v1.0.0-spark, which contains the contents of @ScrapCodes's branch plus an additional patch to configure the build for artifact publication.

/cc @srowen @ScrapCodes for review.

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -21,3 +21,14 @@ libraryDependencies += "org.ow2.asm"  % "asm" % "5.0.3"
...
 addSbtPlugin("com.simplytyped" % "sbt-antlr4" % "0.7.11")
+
+// Spark uses a custom fork of the sbt-pom-reader plugin which contains a patch to fix issues
+// related to test-jar dependencies (https://github.com/sbt/sbt-pom-reader/pull/14). The source for
+// this fork is published at https://github.com/JoshRosen/sbt-pom-reader/tree/v1.0.0-spark
+// and corresponds to commit b160317fcb0b9d1009635a7c5aa05d0f3be61936 in that repository.
+// In the long run, we should try to merge our patch upstream and switch to an upstream version of
+// the plugin; this is tracked at SPARK-14401.
+
+resolvers += "Spark fork of sbt-pom-reader" at "https://oss.sonatype.org/content/repositories/orgspark-project-1124"
```
Comment (rxin) [inline project/plugins.sbt]: is this permanent?

Comment (JoshRosen) [inline project/plugins.sbt]: No, if we decide that the bits that I published here look okay then I'll push to central and remove this resolver.

Comment (rxin) [inline project/plugins.sbt]: please go ahead!

Comment (JoshRosen) [inline project/plugins.sbt]: Done. Let's wait for it to hit Central.

**Comment by ScrapCodes** [comment]
Thanks for doing this. It will definitely save time, and some CPU cycles(energy) for all the first time builds.

**Comment by JoshRosen** [comment]
This is now sync'd to Maven Central and has passed tests, so I'm going to merge it into master, branch-2.0, and branch-1.6.

**Pull request closed without merging**